### PR TITLE
MM-61830 - hide actions to no permissions channels

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR adds the logic to hide the action buttons to non-admin users on scheduled post that were created on channels that the user does not have permissions to.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-61830

#### Screenshots
Before:

https://github.com/user-attachments/assets/f423260f-26ac-4df5-89f0-1710beef840d

After:

https://github.com/user-attachments/assets/9e6c96ae-20ad-4054-905d-99a026edf64b

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/drafts/drafts_illustration.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import React from 'react';
 export default (
     <svg
         width='100%'
-        height='auto'
+        height='100%'
```
Comment (pvev) [inline webapp/channels/src/components/drafts/drafts_illustration.tsx:9]: not related, fix error shown in console about invalid prop value in svg

File: webapp/channels/src/components/drafts/placeholder_scheduled_post_title/placeholder_scheduled_posts_title.tsx [hunk 2]
```diff
@@ -17,6 +19,13 @@ export default function PlaceholderScheduledPostsTitle({type}: Props) {
         />
     );
 
+    const tooltipText = (
+        <FormattedMessage
+            id='scheduled_posts.row_title_thread.placeholder_tooltip'
+            defaultMessage={'This was scheduled in a private channel you are no member of.'}
```
Comment (harshilsharma63) [inline webapp/channels/src/components/drafts/placeholder_scheduled_post_title/placeholder_scheduled_posts_title.tsx:25]: So the `No destination`  can appear even when a channel is hard deleted somehow. So, lets use a generic error message - `The channel either doesn't exist or you do not have access to it`.

**Review by harshilsharma63 — CHANGES_REQUESTED** [review]
@pvev can you handle just one more case - when you create a scheduled post in a channel and then archive that channel, we show the error code and error indicator in such scheduled posts. In that case as well, can you please hide the edit, send now and reschedule icons?

Everything else looks good and works great locally!

**Comment by pvev** [comment]
done @harshilsharma63 , now covers also scenario for archived channels. Tooltip text was updated too and new unit tests added.


https://github.com/user-attachments/assets/190ebe11-2d2f-4262-98a4-3bf2b415cef4

**Review by harshilsharma63 — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]

File: webapp/channels/src/components/drafts/draft_actions/schedule_post_actions/scheduled_post_actions.tsx [hunk 3]
```diff
@@ -127,13 +131,13 @@ function ScheduledPostActions({scheduledPost, channel, onReschedule, onDelete, o
         }));
     }, [channel, dispatch, onSend, scheduledPost.id]);
 
-    const showEditOption = !scheduledPost.error_code;
-
+    const userChannelMember = channel && Object.prototype.hasOwnProperty.call(myChannelsMemberships, channel?.id);
```
[the remark this answers is not in the document]
Comment (pvev) [reply webapp/channels/src/components/drafts/draft_actions/schedule_post_actions/scheduled_post_actions.tsx:134]: makes sense, good suggestion. will try it and if it works will update the code.

**Comment by pvev** [comment]
@amyblais can we please cherry-pick this one to 10.3?

**Comment by harshilsharma63** [comment]
> @amyblais can we please cherry-pick this one to 10.3?

@pvev I added the cherry pick approved tag, as soon as you merge the PR the cherrypick PR wil open up. You can merge this anythime now.

**Pull request merged**